### PR TITLE
#2 Create module to comunicate with server

### PR DESCRIPTION
**Pull request by iverbitskyi**

### CODE REVIEW

**Review by Vlad-2109 — DISMISSED** [review]
We have changed one interface for registration with Ivan and everything is ok!

File: src/api/axios.api.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// http://localhost:8001/api/{NeededService}
+
+import axios from 'axios';
+
+export const instance = axios.create({
+  baseURL: 'http://localhost:8001/api',
```
Comment (VitalikMatvieiev) [inline src/api/axios.api.ts:6]: Here I think we should use env variable

File: src/types/types.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+export interface IUserLogin {
+  userId: number;
+  username: string;
+}
+
+export interface IUserRegistration {
+  username: string;
+  email: string;
+  password: string | number;
+  isAdmin: boolean;
```
Comment (VitalikMatvieiev) [inline src/types/types.ts:16]: Bad name, I think i our case it should be instructor, and I'm not sure that we need all of those model right now, it should be extension of this module when we will be writing the part for that

File: src/services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  IResetPassword,
+  IUserData,
+  IUserId,
+  IUser,
+} from '../types/types';
+
+export const AuthService = {
+  async registration(
+    userData: IUserRegistration,
+  ): Promise<IResponseUserData | undefined> {
```
Comment (VitalikMatvieiev) [inline src/services/auth.service.ts:15]: why do we return undefined in each method?

File: src/api/axios.api.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// http://localhost:8001/api/{NeededService}
```
Comment (VitalikMatvieiev) [inline src/api/axios.api.ts:1]: This whole logoc should be moved to auth-service.ts

File: src/services/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import { instance } from '../api/axios.api';
```
Comment (VitalikMatvieiev) [inline src/services/auth.service.ts:1]: let's do file names with '-'
for example auth-service.ts

File: src/services/auth.service.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  ): Promise<IResponseUserData | undefined> {
+    const { data } = await instance.post<IResponseUserData>(
+      'auth/sign-up',
+      userData,
+    );
+    return data;
+  },
+
+  async login(userData: IUserData): Promise<IUser | undefined> {
+    const { data } = await instance.post<IUser>('auth/sign-in', userData);
```
Comment (VitalikMatvieiev) [inline src/services/auth.service.ts:24]: Error Handling: It's important to handle errors that might occur during API requests. Consider adding try-catch blocks to catch any potential errors and handle them appropriately.

File: src/api/axios.api.ts (hunk 1 above)
Comment (iverbitskyi) [reply src/api/axios.api.ts]: Done

File: src/api/axios.api.ts (hunk 4 above)
Comment (iverbitskyi) [reply src/api/axios.api.ts]: Done

File: src/services/auth.service.ts (hunk 3 above)
Comment (iverbitskyi) [reply src/services/auth.service.ts:15]: undefined here will be returned only if an error happened or the data was not received

File: src/services/auth.service.ts (hunk 6 above)
Comment (iverbitskyi) [reply src/services/auth.service.ts:24]: Done

File: src/types/types.ts (hunk 2 above)
Comment (iverbitskyi) [reply src/types/types.ts:16]: I changed isAdmin to isInstructor. I removed the modules that we don't need yet. Also, the names of the types will continue to change

File: src/types/types.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+export interface IResetPasswordRequest {
```
[the remark this answers is not in the document]
Comment (iverbitskyi) [reply src/types/types.ts:19]: Done

File: package-lock.json
[the remark this answers is not in the document]
Comment (VitalikMatvieiev) [reply package-lock.json:1]: and all other looks good, after do rebase Vlad can show how to do that, I've already showed him

[the remark this answers is not in the document]
Comment (iverbitskyi) [reply package-lock.json:1]: done

**Review by Vlad-2109 — APPROVED** [review]

**Review by VitalikMatvieiev — APPROVED** [review]

**Review by VitalikMatvieiev — APPROVED** [review]

**Review by Vlad-2109 — APPROVED** [review]

**Pull request merged**